### PR TITLE
Add functions to setup the readiness and liveness probes.

### PR DESCRIPTION
**Pull request by singularo**

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -968,6 +968,48 @@ public function generateDeploymentConfig(string $name, string $image_stream_tag,
...
+  public function addProbeConfig(&$deployment_config, $probes) {
+    foreach (['liveness', 'readiness'] as $type) {
+      if (!empty($probes[$type])) {
+        $deployment_config['spec']['template']['spec']['containers'][0][$type . 'Probe'] =
+          $this->probeConfig($probes[$type]);
+      }
+    }
+  }
+
+  private function probeConfig($probe) {
```
Comment (pingers) [inline src/Client.php]: Just make it protected... at least it's testable that way.

File: src/Client.php [hunk 2]
```diff
@@ -968,6 +968,48 @@ public function generateDeploymentConfig(string $name, string $image_stream_tag,
...
+            'command' => explode(' ', $probe['parameters']),
+          ],
+        ];
+        break;
+      case 'httpGet':
+        return [
+          'initialDelaySeconds' => 10,
+          'timeoutSeconds' => 10,
+          'httpGet' => [
+            'port' => (int)$probe['port'],
```
Comment (pingers) [inline src/Client.php]: Missing space between the cast and var.

File: src/Client.php [hunk 3]
```diff
@@ -968,6 +968,48 @@ public function generateDeploymentConfig(string $name, string $image_stream_tag,
     return $deploymentConfig;
   }
 
+  public function addProbeConfig(&$deployment_config, $probes) {
```
Comment (pingers) [inline src/Client.php]: Missing method doc - which belongs in the interface... along with the method declaration there ;)

File: src/Client.php (hunk 1 above)
Comment (pingers) [reply src/Client.php]: Missing method doc.

**Comment by singularo** [comment]
Sorted.

**Comment by pingers** [comment]
:ship:

**Pull request merged**